app/models/manageiq/providers/base_manager/manager_refresher.rb

```diff
@@ -78,6 +78,8 @@ def parse_targeted_inventory(ems, _target, inventory)
       # ManagerRefresh::TargetCollection. This way we can do targeted refresh of all queued targets in 1 refresh
       def preprocess_targets
         @targets_by_ems_id.each do |ems_id, targets|
+          ems = ExtManagementSystem.find(ems_id)
+
           if targets.any? { |t| t.kind_of?(ExtManagementSystem) }
             ems             = @ems_by_ems_id[ems_id]
```

Comment:
Could this line just be moved to where you added the new line setting `ems`

Reply:
Yeah definitely